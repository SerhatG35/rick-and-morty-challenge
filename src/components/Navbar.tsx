import { useEffect, useState } from 'react'
import { useHistory } from 'react-router-dom'
import { ROUTES } from '../constants/Routes'
import { Navbar as Navigation, NavbarRoute } from '../styles/styles'

const Navbar = () => {
    const [pageRoute, setPageRoute] = useState<string | undefined>(undefined)
    const history = useHistory()

    useEffect(() => {
        setPageRoute(history.location.pathname)
    }, [history.location])
    return (
        <Navigation>
            <NavbarRoute
                data-testid='navbar-episodes-route'
                background={pageRoute === ROUTES.EPISODES ? '#FFDC85' : '#eee'}
                href={ROUTES.EPISODES}
            >
                By Episodes
            </NavbarRoute>
            <NavbarRoute
                data-testid='navbar-locations-route'
                background={pageRoute === ROUTES.LOCATIONS ? '#FFDC85' : '#eee'}
                href={ROUTES.LOCATIONS}
            >
                By Locations
            </NavbarRoute>
        </Navigation>
    )
}

export default Navbar
