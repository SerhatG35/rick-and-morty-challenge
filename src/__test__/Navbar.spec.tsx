import { render, waitFor } from '@testing-library/react'
import { useInView } from 'react-intersection-observer'
import Navbar from 'src/components/Navbar'

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: jest.fn(),
        location: {},
    }),
}))

jest.unmock('src/service/dataFetch')

const useInViewMock = useInView as jest.Mock

const links = [
    { id: 'navbar-episodes-route', location: '/' },
    { id: 'navbar-locations-route', location: '/locations' },
]

describe('Navbar tests', () => {
    test('Navbar component rendered correctly', async () => {
        const wrapper = render(<Navbar />)
        await waitFor(() => wrapper)

        expect(wrapper).toMatchSnapshot()
        wrapper.unmount()
    })
    test.each(links)('navbar link hrefs are correct', async link => {
        const wrapper = render(<Navbar />)
        await waitFor(() => wrapper)

        expect(wrapper.getByTestId(link.id)).toHaveAttribute('href', link.location)
        wrapper.unmount()
    })
})
