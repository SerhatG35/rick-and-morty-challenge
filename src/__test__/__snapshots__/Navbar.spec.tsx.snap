// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests Navbar component rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-dlnjwi fgzHqQ"
      >
        <a
          class="sc-hKFxyN kWmvFd"
          data-testid="navbar-episodes-route"
          href="/"
        >
          By Episodes
        </a>
        <a
          class="sc-hKFxyN kWmvFd"
          data-testid="navbar-locations-route"
          href="/locations"
        >
          By Locations
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-dlnjwi fgzHqQ"
    >
      <a
        class="sc-hKFxyN kWmvFd"
        data-testid="navbar-episodes-route"
        href="/"
      >
        By Episodes
      </a>
      <a
        class="sc-hKFxyN kWmvFd"
        data-testid="navbar-locations-route"
        href="/locations"
      >
        By Locations
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
